iphone/cli/lib/process-js-task.js

```diff
@@ -0,0 +1,374 @@
+			file = file.replace(/\./g, '_');
+			info.dest = path.join(this.builder.buildAssetsDir, file);
+			this.builder.jsFilesToEncrypt.push(file); // encrypted name
+		}
+
+		return new Promise((resolve) => {
+			try {
+				this.builder.cli.createHook('build.ios.copyResource', this.builder, (from, to, cb) => {
+					const originalContents = fs.readFileSync(from).toString();
+
+					const r = {
+						original: originalContents,
+						contents: originalContents,
+						symbols: []
+					};
+
+					this.builder.cli.createHook('build.ios.compileJsFile', this.builder, (r, from, to, cb2) => {
+						// Read the possibly modified file contents
+						const source = r.contents;
+
+						// If the file didn't change from previous run, return early
+						const currentHash = this.generateHash(source);
+						const previousHash = this.data.contentHashes[from];
+						if (previousHash && previousHash === currentHash) {
+							this.builder.unmarkBuildDirFile(to);
+							return cb2();
+						}
+
+						this.transformAndCopy(source, from, to, () => {
+							this.data.contentHashes[from] = currentHash;
+							cb2();
+						});
+					})(r, from, to, cb);
+				})(info.src, info.dest, resolve);
+			} catch (ex) {
+				ex.message.split('\n').forEach(m => this.logger.error(m));
+				this.logger.error(ex);
+				process.exit(1);
+			}
+		});
+	}
+
+	/**
+	 * Transforms the given JavaScript source by transpiling and minifying it and then copying
+	 * it to the destination path.
+	 *
+	 * @param {String} source JavaScript source
+	 * @param {String} from Path to the file that contains the JavaScript source
+	 * @param {String} to Path where the transformed source should be saved to
+	 * @param {Function} done Callback function
+	 */
+	transformAndCopy(source, from, to, done) {
+		// DO NOT TRANSPILE CODE inside SDK's common folder. It's already transpiled!
+		const transpile = from.startsWith(this.sdkCommonFolder) ? false : this.defaultAnalyzeOptions.transpile;
+		const minify = from.startsWith(this.sdkCommonFolder) ? false : this.defaultAnalyzeOptions.minify;
+		const analyzeOptions = Object.assign({}, this.defaultAnalyzeOptions, {
+			filename: from,
+			minify,
+			transpile,
+		});
+
+		try {
+			const modified = jsanalyze.analyzeJs(source, analyzeOptions);
+			const newContents = modified.contents;
+
+			// we want to sort by the "to" filename so that we correctly handle file overwriting
+			this.data.tiSymbols[to] = modified.symbols;
+
+			const dir = path.dirname(to);
+			fs.ensureDirSync(dir);
+
+			this.logger.debug(__('Copying and minifying %s => %s', from.cyan, to.cyan));
+			fs.writeFileSync(to, newContents);
+			this.builder.jsFilesChanged = true;
+
+			done();
+		} catch (err) {
+			err.message.split('\n').forEach(m => this.logger.error(m));
+			this.logger.error(err.stack);
+			if (err.codeFrame) { // if we have a nicely formatted pointer to syntax error from babel, use it!
+				this.logger.error(err.codeFrame);
+			}
+			process.exit(1);
```

Comment:
again, I think the builder should be the one making the decision how to handle the Errors